configuration/menu.lua

```diff
@@ -0,0 +1,207 @@
+            found = i
+        end
+    end
+    if found ~= 0 then
+        table.remove(processes, found)
+    end
+    moduleSelectorVar(location.x, location.y, selectionBoxWidth, maxheight-5)
+    moduleDeSelectorVar(location.x+selectionBoxWidth+1, location.y, selectionBoxWidth, maxheight-5)
+    renderer.update()
+end
+
+local function load()
+    local file = io.open("/home/NIDAS/configuration/enabledModules", "r")
+    if file ~= nil then
+        configurationData = serialization.unserialize(file:read("*a"))
+        if configurationData ~= nil then
+            for i = 1, #configurationData.processes do
+                activate(configurationData.processes[i].module, configurationData.processes[i].name, configurationData.processes[i].desc)
+            end
+        else
+            configurationData = {}
+        end
+        file:close()
+    end
+end
+
+local currentTab = nil
+local function flush()
+    if currentTab ~= nil then
+        renderer.removeObject(currentTab)
+    end
+    currentTab = nil
+end
+
+--Modules for configuring start here
+local function infoScreen(x, y, width, height, text, title)
+    flush()
+    currentTab = gui.wrappedTextBox(x, y, width, height, text, title)
+    renderer.update()
+end
+
+local function configScreen(x, y, width, height, title, data)
+    flush()
+    currentTab = gui.configMenu(x, y, width, height, title, data)
+    graphics.context().gpu.setActiveBuffer(currentTab)
+    local configurationTab = data.configure(x, y, gui, graphics, renderer, currentTab)
+    currentTab = {currentTab}
+    if configurationTab ~= nil then for i = 1, #configurationTab do table.insert(currentTab, configurationTab[i]) end end
+end
+
+local selector = nil
+local function moduleSelector(x, y, width, height)
+    if selector ~= nil then
+        renderer.removeObject(selector)
+    end
+    local buttons = {}
+    for i = 1, #modules do
+        local onActivation =
+        {
+            {displayName = "Activate",
+            value = activate,
+            args = {modules[i].module, modules[i].name, modules[i].desc}},
+            {displayName = "Info",
+            value = infoScreen,
+            args = {location.x+2*selectionBoxWidth+2, location.y, maxWidth-(location.x+2*selectionBoxWidth+2), maxheight-5, modules[i].desc, modules[i].name}},
+            {displayName = "Configure",
+            value = configScreen,
+            args = {location.x+2*selectionBoxWidth+2, location.y, maxWidth-(location.x+2*selectionBoxWidth+2), maxheight-5, modules[i].name, require(modules[i].module)}}
+        }
+        table.insert(buttons, {name = modules[i].name, desc = modules[i].desc, func = gui.selectionBox, args = {x+width/2, y+i, onActivation}})
+    end
+    selector = gui.multiButtonList(x, y, buttons, width, height, "Available ".."("..math.floor(#buttons)..")")
+end
+moduleSelectorVar = moduleSelector
```

Comment:
What does `Var` mean?

Reply:
Dummy variable

Basically it's just a variable that holds the function that is initialized earlier on so the IDE doesn't complain.